app/models/api_model.php

```diff
@@ -79,7 +102,7 @@
         }
 
         $supported = array_unique(array_values($project->getLocalesMapping($request->query['store'])));
-        $json = array_values(array_intersect($done, $supported));
+        $json = empty($json) ? array_values(array_intersect($done, $supported)) : $json;
```

Comment:
It works and doesn't throw warnings since this statement is true when I get results, but you're right that this looks too risky for no reason.